.azure-pipelines.yml

```diff
@@ -0,0 +1,248 @@
+name: $(Build.SourceBranch)-$(date:yyyyMMdd)$(rev:.r)
+trigger:
+  branches:
+    include:
+    - 'master'
+  tags:
+    include:
+    - '*'
+
+jobs:
+- job: Build
+  strategy:
+    matrix:
+      windows-stable:
+        imageName: 'vs2017-win2016'
+        rustup_toolchain: stable
+      windows-stable-release:
+        imageName: 'vs2017-win2016'
+        rustup_toolchain: stable
+        build_type: release
+      linux-stable-release:
+        imageName: 'ubuntu-16.04'
+        rustup_toolchain: stable
+        build_type: release
+      mac-stable-release:
+        imageName: 'macos-10.14'
+        rustup_toolchain: stable
+        build_type: release
+      linux-stable:
+        imageName: 'ubuntu-16.04'
+        rustup_toolchain: stable
+      mac-stable:
+        imageName: 'macos-10.14'
+        rustup_toolchain: stable
+      mac-beta:
+        imageName: 'macos-10.14'
+        rustup_toolchain: beta
+      mac-nightly:
+        imageName: 'macos-10.14'
+        rustup_toolchain: nightly
+
+  pool:
+    vmImage: $(imageName)
+
+  variables:
+    SCCACHE_DIR: $(Pipeline.Workspace)/.sccache
+    RUSTC_WRAPPER: sccache
+
+  steps:
+    - checkout: self
+      submodules: true
+    - script: |
+        curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain $RUSTUP_TOOLCHAIN
```

Comment:
I might recommend a snippet like https://github.com/rust-lang/git2-rs/blob/1d33b290e034d0d85ac0e4bc0ac64cebfa8cfbc7/ci/azure-install-rust.yml to install Rust since rustup is already installed on some systems and I've seem some weirdness when two rustup installations get overlaid over one another.